src/sync/installation.ts

```diff
@@ -427,5 +471,41 @@ export const processInstallation =
 				jiraHost
 			});
 
-			await doProcessInstallation(app, queues, job, installationId, jiraHost, logger);
-		};
+			if (await booleanFlag(BooleanFlags.USE_DEDUPLICATOR_FOR_BACKFILLING, false, jiraHost)) {
+				const nextTaskDelays: Array<number> = [];
+
+				const result = await deduplicator.executeWithDeduplication("i-" + installationId,
+					() => doProcessInstallation(app, queues, job, installationId, jiraHost, logger, (delay: number) =>
+						nextTaskDelays.push(delay)
+					));
+
+				switch (result) {
+					case DeduplicatorResult.E_OK:
+						logger.info("Job was executed by deduplicator");
+						if (nextTaskDelays.length > 1) {
+							logger.warn("Multiple next jobs were scheduled");
+						}
```

Comment:
Should we deduplicate it here by scheduling only a single task? 

That was the source of problem which required a deduplication job. Hence all these jobs have absolutely the same payload and shared state in the database, we should schedule only a single job here.

- [ ] Schedule only a single job

Reply:
Choosing maximum delay from the list